File: C#/77_Combinations.cs
public class Solution {
    private void FindSequence(List<IList<int>> combinations, List<int> temp, int n, int k, int curr) {
        if (temp.Count == k)
            combinations.Add(new List<int>(temp));       
        for(int i = curr + 1; i <= n; i++) {
            temp.Add(i);
            FindSequence(combinations, temp, n, k, i);
            temp.RemoveAt(temp.Count-1);
        }        
    }
    public IList<IList<int>> Combine(int n, int k) {
        List<IList<int>> ans = new List<IList<int>>();
        FindSequence(ans, new List<int>(), n, k, 0);
        return ans;
    }
}

File: Python/77_Combinations.py
"""
    The plan:
    
        - Every digit we add n options
        - We go in a depth of K
"""

class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        
        ans = []
        
        def DFS(current_arr, index):
            if len(current_arr) == k:
                ans.append(list(current_arr))
                return
            
            for i in range(index, n + 1):
                current_arr.append(i)
                DFS(current_arr, i + 1)
                current_arr.pop()
        
        DFS([], 1)
        
        return ans
        